components/TreasuryAccount/HoldTokensTotalPrice.tsx

```diff
@@ -0,0 +1,48 @@
+import { getMintMetadata } from '@components/instructions/programs/splToken'
+import useInstructions from '@hooks/useInstructions'
+import {
+  formatMintNaturalAmountAsDecimal,
+  numberWithCommas,
+} from '@tools/sdk/units'
+import tokenService from '@utils/services/token'
+import { useEffect, useState } from 'react'
+
+const HoldTokensTotalPrice = () => {
+  const { governedTokenAccounts } = useInstructions()
+  const [totalPrice, setTotalPrice] = useState('')
+  useEffect(() => {
+    async function calcTotalTokensPrice() {
+      const totalPrice = governedTokenAccounts
+        .filter(
+          (x) => typeof x.mint !== 'undefined' && typeof x.token !== 'undefined'
+        )
+        .map((x) => {
+          return (
+            parseFloat(
```

Comment:
You can use `getMintDecimalAmountFromNatural` to get the decimal amount without formatting/parsing strings